exchange/exchange.go

```diff
@@ -399,13 +403,15 @@ func (e *exchange) HoldAuction(ctx context.Context, r *AuctionRequest, debugLog
 		fledge = extraRespInfo.fledge
 		anyBidsReturned = extraRespInfo.bidsFound
 		r.BidderResponseStartTime = extraRespInfo.bidderResponseStartTime
+		if extraRespInfo.seatNonBidBuilder != nil {
+			seatNonBidBuilder = extraRespInfo.seatNonBidBuilder
+		}
```

Comment:
`extraRespInfo.seatNonBidBuilder` will never be nil based on how `getAllBids` works but perhaps we should keep this nil check given the complexity of this code and since we don't have any unit tests for `getAllBids` to ensure it is never nil? @SyntaxNode thoughts?